File: VB/DXSample/ViewModel/CustomersViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DXSample.Common

Namespace DXSample.ViewModel
Public Class CustomersViewModel
    Public Property Content() As String

    Protected Sub New()
        Content = "A representation of CustomersViewModel."
        ViewInjectionManager.Default.RegisterNavigatedEventHandler(Me, Sub() ViewInjectionManager.Default.Navigate(Regions.Navigation, ModuleType.Customers))
    End Sub
    Public Shared Function Create() As CustomersViewModel
        Return ViewModelSource.Create(Function() New CustomersViewModel())
    End Function
End Class
End Namespace

File: CS/DXSample/ViewModel/CustomersViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DXSample.Common;

namespace DXSample.ViewModel {
public class CustomersViewModel {
    public string Content { get; set; }

    protected CustomersViewModel() {
        Content = "A representation of CustomersViewModel.";
        ViewInjectionManager.Default.RegisterNavigatedEventHandler(this, () => {
            ViewInjectionManager.Default.Navigate(Regions.Navigation, ModuleType.Customers);
        });
    }
    public static CustomersViewModel Create() {
        return ViewModelSource.Create(() => new CustomersViewModel());
    }
}
}
